python/ray/tests/test_failure.py

```diff
@@ -727,24 +728,44 @@ def test_warning_for_too_many_nested_tasks(shutdown_only):
     ray.init(num_cpus=num_cpus)
     p = init_error_pubsub()
 
+    remote_wait = Semaphore.remote(value=0)
+    nested_wait = Semaphore.remote(value=0)
+
+    ray.get([
+        remote_wait.locked.remote(),
+        nested_wait.locked.remote(),
+    ])
+
     @ray.remote
     def f():
         time.sleep(1000)
         return 1
 
     @ray.remote
-    def h():
-        time.sleep(1)
+    def h(nested_waits):
+        nested_wait.release.remote()
+        ray.get(nested_waits)
         ray.get(f.remote())
 
     @ray.remote
-    def g():
+    def g(remote_waits, nested_waits):
         # Sleep so that the f tasks all get submitted to the scheduler after
         # the g tasks.
-        time.sleep(1)
-        ray.get(h.remote())
+        remote_wait.release.remote()
```

Comment:
aren't we calling `nested_wait.release.remote()` twice as often as we lock it?

Reply:
There are 12 `g`. We acquired 12 times inside line 763. Each of g releases once, so it is released 12 times. Where did you find it is released twice more?

Reply:
nested wait is release once per `g` (because each g creates one h), so I think it is released correctly. I might be missing something though. Why did you think it was released twice more?

Reply:
nvm i misread the `nested_wait` as `remote_wait`